flink-runtime/src/main/java/org/apache/flink/runtime/execution/ExecutionState.java

```diff
@@ -52,6 +52,9 @@
 
     DEPLOYING,
 
+    /** Restoring last possible valid state of the task if it has it. */
+    RECOVERING,
```

Comment:
for compatibility purposes it could make sense to move this to the very end, to ensure ordinals from previous jobs still map to the correct state.